src/Helpers/IPFS.ts

```diff
@@ -0,0 +1,61 @@
+import * as FormData from 'form-data'
+import fetch, { Response } from 'node-fetch'
+import { prop } from 'ramda'
+import * as str from 'string-to-stream'
+
+import { minutesToMiliseconds } from './Time'
+
+const getHash = prop('Hash')
+
+interface IPFSConfiguration {
+  readonly url?: string
+}
+
+interface FetchOptions {
+  readonly timeout?: number
+}
+
+export const IPFS = ({
+  url = 'http://localhost:5001',
+}: IPFSConfiguration) => {
+  const paths = {
+    add: `${url}/api/v0/add`,
+    cat: `${url}/api/v0/cat`,
+  }
+
+  const addText = ({
+    timeout = minutesToMiliseconds(10),
+  }: FetchOptions = {}) => async (text: string): Promise<string> => {
+    const formData = new FormData()
+
+    formData.append('file', str(text), {
```

Comment:
Are we no longer using an input stream but reading everything into memory?

Reply:
I believe this is taken from the old use (which is still in place, outside the File API).

Reply:
Can one of you refer me to the code this is going to eventually replace? I don't know where this is supposed to be used then.

Reply:
https://github.com/poetapp/node/blob/master/src/StorageWriter/IPFS.ts
and
https://github.com/poetapp/node/blob/master/src/StorageReader/IPFS.ts

Reply:
also in the middle of duplicating `cat` again
https://github.com/poetapp/node/pull/690 which is why we decided to extract the helper.